File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        HashMap<Integer, Integer> index = new HashMap<>();
        int i;
        for (i=0; i<nums.length; i++) {
            if (map.containsKey(target-nums[i])) {
                break;
            }
            map.put(nums[i], 1);
            index.put(nums[i], i);
        }
        return new int[]{index.get(target-nums[i]), i};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)-1):
            tmp = target - nums[i]
            for j in range(i+1, len(nums)):
                if nums[j]==tmp:
                    return [i, j]